### PR TITLE
Remove unused D_GC variable and documentation

### PR DESCRIPTION
**Pull request by mathias-lang-sociomantic**
This doesn't seem to be used anywhere, hence why I targeted v1.x.x. Let me know if you want to re-target.

OT: v1.x.x should probably be merged into v2.x.x (which is 50 commits behind).

### CODE REVIEW

**Comment by nemanja-boric-sociomantic** [comment]
Why not rename it to `DRT_GCOPT`, as it is done in druntime?

**Comment by nemanja-boric-sociomantic** [comment]
Ah, that was removed as well.

**Comment by leandro-lucarella-sociomantic** [comment]
It is used (in D1), but in a very stealth way: https://github.com/sociomantic-tsunami/tangort/blob/v1.x.x/dmd1.conf#L4

That said, since the config file shipped in tangort now also provides a default (this was probably added before `dmd1.conf` supported adding a default value), I guess it's OK to remove the default. The problem is without the `export` line in MakD, you can propagate the environment variable to the child process (`dmd1`) and thus you can't override it via `make`.

We can still use `D_GC=basic make` I think, and that would be propagated to `dmd1`, so we have a workaround (please test :) ), but this would qualify as a breaking change then, so I would favour going to v2.x.x branch (which I will get up to date soon).

**Comment by leandro-lucarella-sociomantic** [comment]
> Why not rename it to DRT_GCOPT, as it is done in druntime?

They are not the same, `D_GC` is just a hack in the `dmd1.conf` shipped with tangort.

**Comment by mathias-lang-sociomantic** [comment]
Didn't we remove the basic GC from tangort long ago though ?

**Comment by leandro-lucarella-sociomantic** [comment]
> Didn't we remove the basic GC from tangort long ago though ?

Oh, [it looks like we did](https://github.com/sociomantic-tsunami/tangort/tree/v1.x.x/src/gc)! I do not recall at all that happened. Then we need to clean the `dmd1.conf` shipped by tangort too :grimacing:

**Comment by leandro-lucarella-sociomantic** [comment]
Still, technically there are visible changes to the user, and since this change is not really buying us nothing, I would target v2.x.x anyway. Better be safe than sorry.

**Comment by leandro-lucarella-sociomantic** [comment]
This also means adding a "Migration Instructions" note in the RELEASE NOTES.

**Comment by leandro-lucarella-sociomantic** [comment]
#56

**Review by leandro-lucarella-sociomantic — CHANGES_REQUESTED** [review]
Needs release notes and blocked by #56.

**Comment by leandro-lucarella-sociomantic** [comment]
Unblocked, #56 merged.

**Comment by mathias-lang-sociomantic** [comment]
Updated

File: RELEASE_NOTES.md [hunk 1]
```diff
@@ -3,3 +3,5 @@ Migration Instructions
 
 * The `Version` alias to `versionInfo` provided by the `Version` module has been removed.
   All remaining usage can be replaced with `versionInfo`.
+
+* The `D_GC` variable available to `Build.mak` has been removed as there is only one GC available.
```
Comment (leandro-lucarella-sociomantic) [inline RELEASE_NOTES.md]: I would rephrase this with something more in the lines of a migration instruction, like:

> * The `D_GC` variable was removed as it wasn't used any more internally. if you made use of it anywhere, you should define it yourself (in `Config.mak` for example) from now on.

**Review by leandro-lucarella-sociomantic — CHANGES_REQUESTED** [review]

**Comment by mathias-lang-sociomantic** [comment]
Updated

**Review by leandro-lucarella-sociomantic — APPROVED** [review]
